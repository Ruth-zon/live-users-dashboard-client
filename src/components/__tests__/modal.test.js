import { fireEvent, render, screen } from "@testing-library/react";
import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import Modal from '../modal';


describe('<Modal /> test suite', () => {
    const onClose = jest.fn();
  const setup = () => {
    const utils = render(
        <Modal onClose={onClose} show={true}>text</Modal>
    );
    const modal = screen.getByTestId('modal');
    return {
      modal,
      ...utils,
    };
  };
  it('should match snapshot', () => {
    const { asFragment } = setup({ checked: false });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders title', () => {
    const { modal } = setup({ checked: true });
    const title = screen.getByText('Details');
    expect(title).toBeInTheDocument();
  });

  it('renders button', () => {
    const { modal } = setup({ checked: false, indeterminate: true });
    const button = screen.getByText('close');
    expect(button).toBeInTheDocument();
  });

  it('close modal on click', () => {
    const { modal } = setup({ checked: false, indeterminate: true });
    const button = screen.getByText('close');
    fireEvent.click(button);
    expect(onClose).toHaveBeenCalled();
  });

});
