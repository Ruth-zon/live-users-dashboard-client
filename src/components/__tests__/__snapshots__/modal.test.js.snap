// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> test suite should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB guDQoi"
  >
    <div
      class="sc-gTRrQi jovHXp"
      data-testid="modal"
    >
      <h2>
        Details
      </h2>
      <div
        class="content"
      >
        text
      </div>
      <div
        class="actions"
      >
        <button
          class="toggle-button"
        >
          close
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Modal /> test suite should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB guDQoi"
  >
    <div
      class="sc-gTRrQi jovHXp"
      data-testid="modal"
    >
      <h2>
        Details
      </h2>
      <div
        class="content"
      >
        text
      </div>
      <div
        class="actions"
      >
        <button
          class="toggle-button"
        >
          close
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
